### PR TITLE
Add S3 support.

### PR DESCRIPTION
**Pull request by TomaszKolek**

### CODE REVIEW

File: budgetportal/settings.py [hunk 1]
```diff
@@ -150,6 +151,15 @@
 AWS_S3_ENDPOINT_URL = os.environ.get("AWS_S3_ENDPOINT_URL", None)
 AWS_S3_REGION_NAME = os.environ.get("AWS_S3_REGION_NAME", None)
 
+if DEBUG:
+    AWS_DOMAIN = "localhost:9000"
```
Comment (jbothma) [inline budgetportal/settings.py]: let's rather pass values in using per-variable environment variables. `if DEBUG` locks you in a bit and can have surprising effects

Comment (TomaszKolek) [reply budgetportal/settings.py]: Do you want to pass all of those values as env ariale?

File: budgetportal/tests/test_guides_pages.py [hunk 2]
```diff
@@ -33,5 +33,12 @@ def test_guide_page(self):
         """Simple test of template response for guide page"""
         response = Client().get(self.guide_page.url_path)
         self.assertContains(response, self.guide_page.title)
+
+        image_url_without_params = self.guide_page.image.file.url.split("?")[0]
+        # those modifications are done dynamically
+        image_url = image_url_without_params.replace("original_", "")
+        image_url = image_url.replace(".jpg", ".max-320x200.jpg")
+        self.assertContains(response, image_url)
```
Comment (jbothma) [inline budgetportal/tests/test_guides_pages.py]: rather hardcode the image URL and check that it's present in the response, e.g the rough guess below.
Please take the comment varbatim


```suggestion
        ## Verify the integration of our configuration, django-storages and wagtail
        ## - that is - the generated image URL templated in matches the configuration of the site
        self.assertContains(response, "http://minio:9000/images/some-image.max-320x200.jpg")
```

File: budgetportal/tests/test_guides_pages.py [hunk 3]
```diff
@@ -32,6 +34,28 @@ def setUp(self):
     def test_guide_page(self):
         """Simple test of template response for guide page"""
         response = Client().get(self.guide_page.url_path)
```
Comment (jbothma) [inline budgetportal/tests/test_guides_pages.py]: it'd help if you create the image before loading the page :)

File: budgetportal/tests/test_guides_pages.py [hunk 4]
```diff
@@ -32,6 +34,28 @@ def setUp(self):
...
+            access_key=settings.AWS_ACCESS_KEY_ID,
+            secret_key=settings.AWS_SECRET_ACCESS_KEY,
+            secure=False,
+        )
+
+        minio_client.fput_object(
+            settings.AWS_STORAGE_BUCKET_NAME,
+            "original_images/photo.max-320x200.jpg",
+            "budgetportal/tests/test_data/photo.jpg",
+        )
```
Comment (jbothma) [inline budgetportal/tests/test_guides_pages.py]: rather use 
```
from django.core.files.images import ImageFile
from wagtail.wagtailimages.models import Image

image = Image(
    title="Image title",

    # image_file is your StringIO/BytesIO object
    file=ImageFile(image_file, name="image-filename.jpg"),
)
image.save()

Django will automatically place your image file into the media folder.
```

File: budgetportal/templates/base.html [hunk 5]
```diff
@@ -28,7 +28,8 @@
           'self'
           https://vulekamali.gov.za
           https://data.vulekamali.gov.za
-          https://discussions.vulekamali.gov.za;
+          https://discussions.vulekamali.gov.za
+          https://budgetportal-storage.s3.amazonaws.com;
```
Comment (jbothma) [inline budgetportal/templates/base.html]:
```suggestion
```

Comment (jbothma) [inline budgetportal/templates/base.html]:
```suggestion
          https://discussions.vulekamali.gov.za;
```

**Pull request merged**